stdlib/public/core/Codable.swift

```diff
@@ -2894,13 +2894,28 @@ public extension RawRepresentable where RawValue == String, Self : Decodable {
 //===----------------------------------------------------------------------===//
 
 // FIXME: Uncomment when conditional conformance is available.
-extension Array : Codable /* where Element : Codable */ {
-    public init(from decoder: Decoder) throws {
-        guard Element.self is Decodable.Type else {
-            preconditionFailure("Array<\(Element.self)> does not conform to Decodable because \(Element.self) does not conform to Decodable.")
+extension Array : Encodable /* where Element : Encodable */ {
```

Comment:
@airspeedswift Sure, though I think that goes with the point above — we'd prefer this to be the behavior. This is a stopgap fix until we get conditional conformance; the source-breaking change is unfortunate, but I think necessary.

For what it's worth, since the collections are `Encodable` and `Decodable`, I think it'd be somewhat rare to accept one directly in order to encode instead of allowing it to fall into a more generic `encode<T : Encodable>` case, no?